airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/catalog.py

```diff
@@ -0,0 +1,161 @@
+    metadata_data_copy = copy.deepcopy(metadata_data)
+
+    # Extract the override_catalog dictionary
+    override_catalog = metadata_data_copy["catalogs"][override_catalog_key]
+
+    # Remove the "enabled" key from the override_catalog dictionary
+    del override_catalog["enabled"]
+
+    # Update the metadata_data_copy dictionary with the values from the override_catalog dictionary
+    metadata_data_copy.update(override_catalog)
+
+    return metadata_data_copy
+
+
+def metadata_to_catalog_entry(metadata_definition: dict, connector_type: str, override_catalog_key: str) -> dict:
+    # Make a deep copy of the input dictionary to avoid side effects
+    metadata_definition_copy = copy.deepcopy(metadata_definition)
+    metadata_data = metadata_definition_copy["data"]
+
+    # remove the metadata fields that were added
+    overrode_metadata_data = apply_overrides_from_catalog(metadata_data, override_catalog_key)
+    del overrode_metadata_data["catalogs"]
+
+    # remove connectorType field
+    del overrode_metadata_data["connectorType"]
+
+    # rename field connectorSubtype to sourceType
+    connection_type = overrode_metadata_data.get("connectorSubtype")
+    if connection_type:
+        overrode_metadata_data["sourceType"] = overrode_metadata_data["connectorSubtype"]
+        del overrode_metadata_data["connectorSubtype"]
+
+    # rename supportUrl to documentationUrl
+    support_url = overrode_metadata_data.get("supportUrl")
+    if support_url:
+        overrode_metadata_data["documentationUrl"] = overrode_metadata_data["supportUrl"]
+        del overrode_metadata_data["supportUrl"]
+
+    # rename definitionId field to sourceDefinitionId or destinationDefinitionId
+    id_field = "sourceDefinitionId" if connector_type == "source" else "destinationDefinitionId"
+    overrode_metadata_data[id_field] = overrode_metadata_data["definitionId"]
+    del overrode_metadata_data["definitionId"]
+
+    # add in useless fields that are currently required for porting to the actor definition spec
+    overrode_metadata_data["tombstone"] = False
+    overrode_metadata_data["custom"] = False
+    overrode_metadata_data["public"] = True
+
+    # if there is no releaseStage, set it to "alpha"
+    # Note: this is something our current cloud catalog generator does
+    # Note: We will not once this is live
+    if not overrode_metadata_data.get("releaseStage"):
+        overrode_metadata_data["releaseStage"] = "alpha"
+
+    return overrode_metadata_data
+
+
+def is_metadata_catalog_enabled(metadata_definition: dict, catalog_name: str) -> bool:
+    return metadata_definition["data"]["catalogs"][catalog_name]["enabled"]
+
+
+def is_metadata_connector_type(metadata_definition: dict, connector_type: str) -> bool:
+    return metadata_definition["data"]["connectorType"] == connector_type
+
+
+def construct_catalog_from_metadata(catalog_derived_metadata_definitions: List[dict], catalog_name: str) -> dict:
```

Comment:
```suggestion
def construct_catalog_from_metadata(catalog_derived_metadata_definitions: List[PartialMetadataDefinition], catalog_name: str) -> dict:
```

I'm not sure if you're passing a list of dict or PartialMetadataDefinition because `cloud_catalog_from_metadata` passes its `catalog_derived_metadata_definitions` input list composed of PartialMetadataDefinition.